src/Board.hs

```diff
@@ -60,14 +62,14 @@ scrambleBoard board@(Board (Dimensions width height) _) startingBoard gen =
                in verifyBoard newBoard gen iterations
           verifyBoard board gen iterations
               | board == startingBoard = scrambleBoardLoop board gen iterations
-              | otherwise              = scrambleBoardLoop board gen (pred iterations)
+              | otherwise              = scrambleBoardLoop board gen (iterations - 1)
 
 -- | Randomly generate a new board, given dimensions.
 newBoard :: Dimensions -> IO Board
 newBoard d = do
     newGen <- R.newStdGen
-    return $ scrambleBoard startingBoard startingBoard newGen
-    where startingBoard = calculateStartingBoard $ Board d []
+    return (scrambleBoard startingBoard startingBoard newGen)
+    where startingBoard = calculateStartingBoard (Board d [])
```

Comment:
I usually only use `$` when the only alternative is having 2 or 3 levels of parenthesis. If there's only one level, then I use parenthesis. Some people try to not use `$` at all. I was heavily influenced by http://www.haskellforall.com/2015/09/how-to-make-your-haskell-code-more.html. The argument against using `$` is that it makes it harder for Haskell beginners or non-Haskellers to understand, and that it misleads non-Haskellers to believe that Haskell depends heavily on confusing operator syntax. On the other hand, the meaning of parenthesis is obvious to most programmers.